Core/src/commands/player/ReportCommand.ts

```diff
@@ -227,6 +236,56 @@ async function handleInnRoomReaction(
 	}));
 }
 
+async function handleEnchantReaction(player: Player, reaction: ReactionCollectorEnchantReaction, response: CrowniclesPacket[]): Promise<void> {
+	const enchantment = ItemEnchantment.getById(await Settings.ENCHANTER_ENCHANTMENT_ID.getValue());
+	const isPlayerMage = player.class === ClassConstants.CLASSES_ID.MYSTIC_MAGE;
+	const price = enchantment.getEnchantmentCost(isPlayerMage);
+
+	const hasEnoughMoney = player.money >= price.money;
+	const playerMissionsInfo = price.gems !== 0 ? await PlayerMissionsInfos.getOfPlayer(player.id) : null;
+	const hasEnoughGems = playerMissionsInfo ? playerMissionsInfo.gems >= price.gems : true;
+
+	if (!hasEnoughMoney || !hasEnoughGems) {
+		response.push(makePacket(CommandReportEnchantNotEnoughCurrenciesRes, {
+			missingMoney: hasEnoughMoney ? 0 : price.money - player.money,
+			missingGems: hasEnoughGems ? 0 : price.gems - playerMissionsInfo.gems
+		}));
+		return;
+	}
+
+	const itemToEnchant = await InventorySlots.getItem(player.id, reaction.slot, reaction.itemCategory);
+	if (!itemToEnchant || !(itemToEnchant.isWeapon() || itemToEnchant.isArmor()) || itemToEnchant.itemEnchantmentId) {
+		CrowniclesLogger.error("Player tried to enchant an item that doesn't exist or cannot be enchanted. It shouldn't happen because the player must not be able to switch items while in the collector.");
+		response.push(makePacket(CommandReportItemCannotBeEnchantedRes, {}));
+		return;
+	}
+
+	await player.reload();
+
+	itemToEnchant.itemEnchantmentId = enchantment.id;
+	if (price.money > 0) {
+		await player.spendMoney({
+			response,
+			amount: price.money,
+			reason: NumberChangeReason.ENCHANT_ITEM
+		});
+	}
+	if (price.gems > 0) {
+		await playerMissionsInfo.spendGems(price.gems, response, NumberChangeReason.ENCHANT_ITEM);
+	}
+
+	await Promise.all([
+		itemToEnchant.save(),
+		player.save(),
+		playerMissionsInfo ? playerMissionsInfo.save() : Promise.resolve()
```

Comment:
et du coup on peut supprimer la ternaire aussi ici, ca coute rien un save
```suggestion
		playerMissionsInfo.save()
```

Reply:
Avec sequelize si